client/components/tinymce/plugins/wpcom-track-paste/plugin.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+import debugFactory from 'debug';
+import tinymce from 'tinymce/tinymce';
+import { recordTracksEvent } from 'state/analytics/actions';
+
+const debug = debugFactory( 'calypso:tinymce-plugins:wpcom-paste-track' );
+
+const SOURCE_GOOGLE_DOCS = 'google_docs';
+const SOURCE_UNKNOWN = 'unknown';
+
+function trackPaste( editor ) {
+	debug( 'init' );
+
+	const store = editor.getParam( 'redux_store' );
+	if ( ! store ) {
+		debug( 'store not found, do not bind events' );
+		return;
```

Comment:
> if Redux is missing there is a bigger problem - and my feeling is that it is the editor responsibility to fix it, not plugins

in this case, why test to make sure that it's there?

the questions I would ask are as follows:
 - do we expect the store to be there sometimes and not always? if so, should this plugin simply not load if it's missing? if so, `return` is fine
 - do we require that the store be there always? if so, do we expect some other code to be responsible for loading it? if so, we should either throw (or maybe more appropriately) just _assume_ that it's there. if it's not there we will end up `throw`ing an error anyway when we try to dispatch an action.

personally I would be in favor of removing this code: it's redundant if we really believe that the store should be here; it's confusing because it leads to the question "when would this happen and what should happen?"